Add max_connections to TCP input

kvch opened the pull request:
New configuration option is added to the TCP input of Filebeat to limit the number of incoming connections. By default it is set to 0 which means, it's unlimited.

```
# Max number of concurrent connections, or 0 for no limit. Default: 0
#max_connections: 0
```

Vendored
* `golang.org/x/net/netutil`

michalpristas commented:
this looks ok, what i'm thinking about is whether or not is this solving root cause.
issue mentioned that by introducing weird networking issue we're running out of fd's in the system. this means, that connections are not getting closed properly or are created in much higher speed than closed. 
if later is the case this PR will fix that, but if we have a problem with closing connections, LimitListener will just block any new connections from  being opened. not sure if this makes sense

faec commented on filebeat/input/tcp/config.go:
```diff
@@ -40,6 +40,7 @@ var defaultConfig = config{
 	Config: tcp.Config{
 		Timeout:        time.Minute * 5,
 		MaxMessageSize: 20 * humanize.MiByte,
+		MaxConnections: 10,
```
What are the effects of this default value on a typical node? Are we confident that this is high enough not to artificially bottleneck a node that was previously handling a higher load? I'd be happier if this value was higher, or if the default was "no maximum"

kvch replied:
Good point! Updated the configuration to be unlimited by default.

kvch commented:
@michalpristas I have addressed your concerns in the relevant issue. It is indeed a workaround, proper solution, if it exists at all, needs more investigation.

faec commented on filebeat/_meta/common.reference.inputs.yml:
```diff
@@ -231,8 +231,8 @@ filebeat.inputs:
-  # Max number of concurrent connections. Default: 10
-  #maxconn: 10
+  # Max number of concurrent connections. By default it is not limited.
+  #maxconn: 0
```
It looks like "maxconn" doesn't exist in the configuration, was this supposed to be removed?

Also make the documentation more explicit (after moving it below), e.g. "Max number of concurrent connections, or 0 for no limit. Default: 0"

kvch replied:
:man_facepalming:

faec commented on filebeat/input/tcp/config.go:
```diff
@@ -40,6 +40,7 @@ var defaultConfig = config{
 	Config: tcp.Config{
 		Timeout:        time.Minute * 5,
 		MaxMessageSize: 20 * humanize.MiByte,
+		MaxConnections: 0,
```
It might be simpler to omit this line since 0 is already the default

kvch commented:
Failing tests are unrelated.
